tensorboard/webapp/widgets/card_fob/card_fob_controller_test.ts

```diff
@@ -398,6 +418,51 @@ describe('card_fob_controller', () => {
       ).toEqual(3);
       expect(onTimeSelectionChanged).toHaveBeenCalledTimes(0);
     });
+
+    it('adds and removes event listener', () => {
+      const fixture = createComponent({
+        timeSelection: {start: {step: 4}, end: null},
+      });
+      const addListenerSpy = jasmine.createSpy();
+      const removeListenerSpy = jasmine.createSpy();
+      document.addEventListener = addListenerSpy;
+      let mouseMoveListener;
+      let mouseUpListener;
+      addListenerSpy.and.callFake((type, listener) => {
+        if (type === 'mousemove') {
+          mouseMoveListener = listener;
+        }
+
+        if (type === 'mouseup') {
+          mouseUpListener = listener;
+        }
+      });
+      document.removeEventListener = removeListenerSpy;
```

Comment:
This test has side effects that could impact other tests - modifications to document.addEventListener and document.removeEventListener. This could cause subtle problems in other tests. My recommendation would be to modify document.addEventListener and document.removeEventListener in the beforeEach() method and then clean them back up in the afterEach() method.

Reply:
Not sure how to clean up in afterEach. However, I started using spyOn which from what I understand automatically removes the spy after the test.